isis/src/base/apps/ddd2isis/ddd2isis.cpp

```diff
@@ -41,86 +33,142 @@ void IsisMain() {
    *
    */
 
-  // Verify the magic number
+   // ifstream read() needs a char* to read values into, so the union
+   // is used to store read values
+   union {
+     char readChars[4];
+     long readLong;
+     float readFloat;
+   } readBytes;
+
+   // ddd files are LSB
+   EndianSwapper swp("MSB");
+
+  // Verify that the file is a ddd by reading in the first 4 bytes and
+  // comparing the magic numbers
   readBytes.readLong = 0;
   fin.seekg(0);
   fin.read(readBytes.readChars, 4);
+  if( fin.fail() || fin.eof() ) {
+    string msg = "Could not read the magic number in the input file [" + from + "]";
+    throw IException(IException::Io, msg, _FILEINFO_);
+  }
   readBytes.readFloat = swp.Float(readBytes.readChars);
 
-  if(readBytes.readLong != 0x67B) {
+  if(readBytes.readLong != 1659) {
```

Comment:
1659 seems like a magic number. Can you add a comment to explain it?

Reply:
I added on to the comment above that chunk of code explaining the magic numbers.

Reply:
I talked to Stuart about clarifying who we got the magic number from, and he said not to put anyone's info in the comments. For now, just specifying that 1659 is the number that identifies a ddd is okay.